java-checks/src/main/java/org/sonar/java/filters/LombokFilter.java

```diff
@@ -181,12 +182,34 @@ public void visitAssignmentExpression(AssignmentExpressionTree tree) {
   }
 
   private static boolean usesAnnotation(ClassTree classTree, List<String> annotations) {
+    return usesAnnotation(classTree, annotations, !CHECK_LOCAL_NAME);
+  }
+
+  private static final boolean CHECK_LOCAL_NAME = true;
+
+  private static boolean usesAnnotation(ClassTree classTree, List<String> annotations, boolean checkLocalName) {
     SymbolMetadata classMetadata = classTree.symbol().metadata();
-    return annotations.stream().anyMatch(classMetadata::isAnnotatedWith);
+
+    for(String fullyQualified: annotations) {
+      if(classMetadata.isAnnotatedWith(fullyQualified)) {
+        return true;
+      }
+      // In automatic analysis use only the last part of the annotation.
+      if(checkLocalName && classMetadata.isAnnotatedWith(localName(fullyQualified))) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  @VisibleForTesting
+  static String localName(String fullyQualified) {
```

Comment:
Nitpick: what about a more [specific name](https://docs.oracle.com/javase/specs/jls/se21/html/jls-6.html#jls-TypeName)?
```suggestion
  static String annotationTypeIdentifier(String fullyQualified) {
```